File: lib/src/main/java/vserver/route/WildcardSubPath.java
package vserver.route;

import vserver.SubPath;
import vserver.RoutingContext;

public class WildcardSubPath implements SubPath {
    private final SubPath next;

    public WildcardSubPath(SubPath next) {
        this.next = next;
    }

    @Override
    public SubPath next() {
        return next;
    }

    @Override
    public boolean match(String route) {
        return true;
    }

    @Override
    public boolean currentSame(SubPath r) {
        return r instanceof WildcardSubPath;
    }

    @Override
    public void fill(RoutingContext ctx, String route) {
    }
}

File: lib/src/main/kotlin/io/vproxy/lib/http/route/WildcardSubPath.kt
package io.vproxy.lib.http.route

import io.vproxy.lib.http.RoutingContext

class WildcardSubPath(private val next: SubPath?) : SubPath {
  override fun next(): SubPath? {
    return next
  }

  override fun match(route: String): Boolean {
    return true
  }

  override fun currentSame(r: SubPath): Boolean {
    return r is WildcardSubPath
  }

  override fun fill(ctx: RoutingContext, route: String) {}

  override fun toString(): String {
    return "/*"
  }
}
